bugzooka/integrations/rag_client_util.py

```diff
@@ -0,0 +1,89 @@
+import os
+from typing import Optional
+
+from dotenv import load_dotenv
+from llama_index.core import Settings, load_index_from_storage
+from llama_index.core.llms.utils import resolve_llm
+from llama_index.core.storage.storage_context import StorageContext
+from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+from llama_index.vector_stores.faiss import FaissVectorStore
+
+# Fix cache permission issue for non-root containers
+os.environ.setdefault("HF_HOME", "/tmp/.cache")
+os.environ.setdefault("TRANSFORMERS_CACHE", "/tmp/.cache")
+os.environ.setdefault("LLAMA_INDEX_CACHE_DIR", "/tmp/.cache")
+os.makedirs("/tmp/.cache", exist_ok=True)
+
+
+def get_rag_context(query: str, top_k: Optional[int] = None) -> str:
+    """Return concatenated top-k chunks from the local FAISS store for a query.
+
+    Reads configuration from environment variables and optional .env files.
+    """
+    # Load env without overriding already-set variables
+    load_dotenv(dotenv_path=".env", override=False)
+    load_dotenv(dotenv_path="/app/.env", override=False)
+
+    db_path = os.getenv("RAG_DB_PATH", "/rag")
+    index_id = os.getenv("RAG_PRODUCT_INDEX", "vector_db_index")
+    embed_model_path = os.getenv(
```

Comment:
How big is this model? Does it add too much delay while embedding text? Shall we consider any of our hosting offerings like MAAS or models.corp for this?

Reply:
we want to use the same model used for creating embeddings https://github.com/openshift/lightspeed-rag-content/blob/d70c7326f6aaeb7b7fe52e154cff5463f664c5c1/byok/Containerfile.output#L12